test/IDE/complete_unresolved_chains.swift

```diff
@@ -0,0 +1,79 @@
+// RUN: %target-swift-ide-test -code-completion -source-filename %s -code-completion-token=UNRESOLVED_CHAIN_1 | %FileCheck %s -check-prefix=UNRESOLVED_CHAIN_1
+// RUN: %target-swift-ide-test -code-completion -source-filename %s -code-completion-token=UNRESOLVED_CHAIN_2 | %FileCheck %s -check-prefix=UNRESOLVED_CHAIN_1
+// RUN: %target-swift-ide-test -code-completion -source-filename %s -code-completion-token=UNRESOLVED_CHAIN_3 | %FileCheck %s -check-prefix=UNRESOLVED_CHAIN_1
+// RUN: %target-swift-ide-test -code-completion -source-filename %s -code-completion-token=UNRESOLVED_CHAIN_4 | %FileCheck %s -check-prefix=UNRESOLVED_CHAIN_1
+// RUN: %target-swift-ide-test -code-completion -source-filename %s -code-completion-token=UNRESOLVED_CHAIN_5 | %FileCheck %s -check-prefix=UNRESOLVED_CHAIN_2
+// RUN: %target-swift-ide-test -code-completion -source-filename %s -code-completion-token=UNRESOLVED_CHAIN_6 | %FileCheck %s -check-prefix=UNRESOLVED_CHAIN_3
+// RUN: %target-swift-ide-test -code-completion -source-filename %s -code-completion-token=UNRESOLVED_CHAIN_7 | %FileCheck %s -check-prefix=UNRESOLVED_CHAIN_3
+// RUN: %target-swift-ide-test -code-completion -source-filename %s -code-completion-token=DOUBLY_NESTED| %FileCheck %s -check-prefix=DOUBLY_NESTED
+
+struct ChainStruct1 {
+  static var chain2 = ChainStruct2()
+  static func chain2Func() -> ChainStruct2 { ChainStruct2() }
+}
+
+struct ChainStruct2 {
+  var chainStruct1 = ChainStruct1()
+  var chainEnum = ChainEnum.case1
+  var chainStruct2: ChainStruct2 { ChainStruct2() }
+  func chainStruct1Func() -> ChainStruct1 { ChainStruct1() }
+}
+
+enum ChainEnum {
+  case case1
+  var chainStruct2: ChainStruct2 { ChainStruct2() }
+  func chainStruct2Func() -> ChainStruct2 { ChainStruct2() }
+}
+
+func testChains() {
+  let _: ChainStruct1 = .chain2.#^UNRESOLVED_CHAIN_1^#
+  let _: ChainStruct1 = .chain2.chainStruct2.#^UNRESOLVED_CHAIN_2^#
+  let _: ChainStruct1 = .chain2Func().#^UNRESOLVED_CHAIN_3^#
+  let _: ChainStruct1 = .chain2Func().#^UNRESOLVED_CHAIN_4^#
+  let _: ChainEnum = .case1.#^UNRESOLVED_CHAIN_5^#
+  let _: ChainEnum = .case1.chainStruct2.#^UNRESOLVED_CHAIN_6^#
+  let _: ChainEnum = .case1.chainStruct2.#^UNRESOLVED_CHAIN_7^#
+}
+
+// UNRESOLVED_CHAIN_1: Begin completions, 5 items
```

Comment:
Why all of the completions declared that there are 5 items but checks list maximum four?

Reply:
Ah that's because I omitted the `.self` completions, can add those back if you think it's worth it to check that those are offered.

Reply:
Let's include everything to make sure that in the future nothing would get excluded unnoticed...